pkg/auth/providers/googleoauth/goauth_helper.go

```diff
@@ -205,6 +216,33 @@ func (g *googleOauthProvider) paginateResults(adminSvc *admin.Service, hostedDom
 	}
 }
 
+func (g *googleOauthProvider) gatherParentGroups(groupPrincipal v3.Principal, adminSvc *admin.Service, config *v3.GoogleOauthConfig, hostedDomain string, groupMap map[string]bool, nestedGroupPrincipals *[]v3.Principal) error {
+	if groupMap[groupPrincipal.ObjectMeta.Name] {
+		return nil
+	}
+	groupMap[groupPrincipal.ObjectMeta.Name] = true
+	parts := strings.SplitN(groupPrincipal.ObjectMeta.Name, ":", 2)
+	if len(parts) != 2 {
+		return fmt.Errorf("invalid id %v", groupPrincipal.ObjectMeta.Name)
```

Comment:
Can we have an error message with more information, like maybe "error while gathering nested groups, invalid group id"